tests/codegen/waiter-tests/src/test/kotlin/com/test/WaiterTest.kt

```diff
@@ -5,90 +5,774 @@
 
 package com.test
 
-import aws.smithy.kotlin.runtime.ServiceErrorMetadata
-import aws.smithy.kotlin.runtime.ServiceException
 import aws.smithy.kotlin.runtime.retries.getOrThrow
-import aws.smithy.kotlin.runtime.InternalApi
-import com.test.model.GetEntityResponse
-import com.test.waiters.waitUntilEntityExistsBySuccess
-import com.test.waiters.waitUntilEntityHasComparableNumericalValues
-import com.test.waiters.waitUntilEntityHasNameTagByOutput
-import kotlinx.coroutines.runBlocking
-import kotlin.test.Test
-import kotlin.test.assertEquals
-import kotlin.test.assertFailsWith
+import aws.smithy.kotlin.runtime.retries.Outcome
+import com.test.model.*
+import com.test.model.Enum
+import com.test.waiters.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.*
 
+@OptIn(ExperimentalCoroutinesApi::class)
 class WaiterTest {
-    @OptIn(InternalApi::class)
-    private fun failure(errorCode: String) = Result.failure<GetEntityResponse>(
-        ServiceException().apply {
-            sdkErrorMetadata.attributes[ServiceErrorMetadata.ErrorCode] = errorCode
-        }
-    )
-
-    private fun success(builderBlock: GetEntityResponse.Builder.() -> Unit) =
-        GetEntityResponse.Builder().apply(builderBlock).build().let { success(it) }
+    private fun successTest(
+        results: List<GetEntityResponse>,
+        block: suspend WaitersTestClient.(request: GetEntityRequest) -> Outcome<GetEntityResponse>,
+    ): Unit = runTest {
```

Comment:
**Suggestion**: Putting the `results` parameter after the `block` parameter and making it `vararg` would simplify your invocation from:

```kotlin
@Test fun testBooleanEquals() = successTest(
    listOf(
        GetEntityResponse { primitives = EntityPrimitives { boolean = false } },
        GetEntityResponse { primitives = EntityPrimitives { boolean = true } },
    ),
    WaitersTestClient::waitUntilBooleanEquals,
)
```

To:

```kotlin
@Test fun testBooleanEquals() = successTest(
    WaitersTestClient::waitUntilBooleanEquals,
    GetEntityResponse { primitives = EntityPrimitives { boolean = false } },
    GetEntityResponse { primitives = EntityPrimitives { boolean = true } },
)
```